core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -75,6 +77,23 @@ public static LocationProvider locationsFor(
     }
   }
 
+  private static final Set<String> DEPRECATED_PROPERTIES =
+      ImmutableSet.of(
+          TableProperties.OBJECT_STORE_PATH, TableProperties.WRITE_FOLDER_STORAGE_LOCATION);
+
+  private static String getProperty(Map<String, String> properties, String key) {
```

Comment:
maybe `checkDeprecation`?

Reply:
`checkDeprecation` is also not accurate. This method actually does both: get property value and check deprecations.

 the implementation/error msg is tied to `TableProperties.WRITE_DATA_LOCATION`. Maybe call this `getLegacyDataLocation` or sth like that.

The other approach is to define `DEPRECATED_PROPERTIES` in `TableProperties` class as a map (key = deprecated property, value = new property). maybe we can get to it when the need comes.  right now, we can keep the change local to `LocationProviders` class

Reply:
Went for `getAndCheckLegacyLocation` since it also applies to metadata. And yes, everything is private, so we can always move to the `TableProperties`.